app/cdash/tests/test_starttimefromupload.php

```diff
@@ -0,0 +1,65 @@
+<?php
+require_once dirname(__FILE__) . '/cdash_test_case.php';
+
+use CDash\Model\Build;
+use CDash\Model\Project;
+use Illuminate\Support\Facades\DB;
+
+class StartTimeFromUploadTestCase extends KWWebTestCase
+{
+    private Project $project;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->project = new Project();
+    }
+
+    public function __destruct()
+    {
+        // Delete project & build created by this test.
+        remove_project_builds($this->project->Id);
+        $this->project->Delete();
+    }
+
+    public function testStartTimeFromUpload() : void
+    {
+        $this->deleteLog($this->logfilename);
+
+        // Create test project.
+        $this->login();
+        $this->project->Id = $this->createProject([
+            'Name' => 'StartTimeFromUpload',
+            'NightlyTime' => '18:00:00 America/Denver',
+        ]);
+        $this->project->Fill();
+
+        // Submit our testing data.
+        $file = dirname(__FILE__) . '/data/StartTimeFromUpload/Upload.xml';
+        if (!$this->submission('StartTimeFromUpload', $file)) {
+            $this->fail("Failed to submit {$file}");
+        }
+
+        // No errors in the log.
+        $this->assertTrue($this->checkLog($this->logfilename) !== false);
+
+        // The build exists.
+        $results = \App\Models\Project::findOrFail((int) $this->project->Id)->builds()->get();
```

Comment:
I believe you could fix the ignored PHPStan error by doing this:
```suggestion
        $results = \App\Models\Project::findOrFail((int) $this->project->Id)->builds()->firstOrFail();
```

Reply:
My latest force push got me as close as I could 🤷 . If I change it to `::firstOrFail()` the test fails with

```
Unexpected exception of type [BadMethodCallException] with message [Method Illuminate\Database\Eloquent\Relations\HasMany::firstOrFail does not exist.]
```

Reply:
That looks better to me.  The static/dynamic method duality magic Laravel does confuses PHPStan and I eventually hope to add some rules to specifically ignore Laravel functions, which should resolve the new error.